resolver/resolver.cc

```diff
@@ -1062,21 +1062,13 @@ class ResolveConstantsWalk {
                        job.klass.data(ctx)->superClass() == core::Symbols::todo() ||
                        job.klass.data(ctx)->superClass() == resolvedClass) {
                 job.klass.data(ctx)->setSuperClass(resolvedClass);
-            } else if (!ctx.file.data(ctx).isPayload() && absl::c_any_of(job.klass.data(ctx)->locs(), [&](auto &loc) {
```

Comment:
I'm not sure the need for this branch, was the goal another pass at this code for the same class? If that's the intent it wasn't happening. Maybe I'm breaking something else by removing this.

Reply:
Sorry for the delay in review here. I think that this is likely just me not thinking straight when I landed this change. I don't remember, but I'm guessing that what happened was that at first I tried to say "you can do this for any payload definition" and then Alexandre asked that it be gated behind a command line option to mention specific classes instead of "all classes in the payload" and I forgot to delete this code when I refactored my change.

Thanks for catching this and submitting a patch.